client/src/modules/home/index.tsx

```diff
@@ -84,22 +66,35 @@ const Home = () => {
             <Heading as="h2" size={'md'}>
               Upcoming events
             </Heading>
-            {data.paginatedEvents.map((event) => (
-              <EventCard key={event.id} event={event} />
-            ))}
-            {hasMore ? (
-              <Button onClick={onLoadMore}>Click for more</Button>
-            ) : (
-              <Text size="md">No more</Text>
-            )}
+            <Pagination
+              data={data.paginatedEventsWithTotal}
+              dataFlattener={(eventsData) => eventsData.events}
+              itemsPerPage={eventsPerPage}
+              mapper={(event: EventCardProps['event']) => (
+                <EventCard key={event.id} event={event} />
+              )}
+              onClickForMore={(offset: number) =>
+                fetchMore({ variables: { offset, limit: eventsPerPage } })
+              }
+              totalItemsFromData={(data) => data[0].total}
+              displayOnEmpty={
+                <Text size="md">
+                  No more, check out the{' '}
+                  <Link href="/events" fontWeight="bold">
+                    Event page
+                  </Link>
+                  .
```

Comment:
```suggestion
                  to see past events.
```
It wasn't obvious to me why I should go there otherwise.